homeassistant/components/smhi/__init__.py

```diff
@@ -21,3 +28,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    if entry.version == 1:
+        new_data = {
+            CONF_NAME: entry.data[CONF_NAME],
+            CONF_LOCATION: {
+                CONF_LATITUDE: entry.data[CONF_LATITUDE],
+                CONF_LONGITUDE: entry.data[CONF_LONGITUDE],
+            },
+        }
+        new_unique_id = f"smhi-{entry.data[CONF_LATITUDE]}-{entry.data[CONF_LONGITUDE]}"
+
+        if update_entry := hass.config_entries.async_update_entry(
+            entry, data=new_data, unique_id=new_unique_id
+        ):
+            entry.version = 2
+
+    if entry.version == 2:
+        new_unique_id_entity = f"smhi-{entry.data[CONF_LOCATION][CONF_LATITUDE]}-{entry.data[CONF_LOCATION][CONF_LONGITUDE]}"
+
+        @callback
+        def update_unique_id(entity_entry: RegistryEntry) -> dict[str, str]:
+            """Update unique ID of entity entry."""
+            return {"new_unique_id": new_unique_id_entity}
+
+        await async_migrate_entries(hass, entry.entry_id, update_unique_id)
+
+    return update_entry
```

Comment:
Can this be rewritten as:
```suggestion
        if not hass.config_entries.async_update_entry(
            entry, data=new_data, unique_id=new_unique_id
        ):
            return False

        new_unique_id_entity = f"smhi-{entry.data[CONF_LOCATION][CONF_LATITUDE]}-{entry.data[CONF_LOCATION][CONF_LONGITUDE]}"

        @callback
        def update_unique_id(entity_entry: RegistryEntry) -> dict[str, str]:
            """Update unique ID of entity entry."""
            return {"new_unique_id": new_unique_id_entity}

        await async_migrate_entries(hass, entry.entry_id, update_unique_id)

    return True
```